python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -180,11 +186,21 @@ def my_asset(my_upstream_asset: int) -> int:
     """
 
     def create_asset():
+        if non_argument_deps is not None and deps is not None:
+            raise DagsterInvalidDefinitionError(
+                "Cannot specify both deps and non_argument_deps. Use only deps instead."
+            )
+
+        upstream_asset_deps: Optional[Set[CoercibleToAssetKey]] = deps
+        if non_argument_deps is not None:
+            deprecation_warning("non_argument_deps", "X.X.X", "use parameter deps instead")
```

Comment:
TODO - set deprecation version

Reply:
2.0.0 i would guess - no reason to hard remove ahead of that i think